File: Java/0709_To_Lower_Case.java
class Solution {
    public String toLowerCase(String s) {
        StringBuilder sb = new StringBuilder(s);

        for(int i = 0; i < s.length(); i++){
            char c = sb.charAt(i);
            if(c >= 'A' && c <= 'Z'){
                c -= ('A' - 'a');
                sb.setCharAt(i, c);
            }
        }
        return sb.toString();
    }
}

File: Cpp/0709_To_Lower_Case.cpp
class Solution {
public:
    string toLowerCase(string s) {
        for(int i = 0; i < s.size(); i++){
            s[i] = (s[i] >= 'A' && s[i] <= 'Z') ? s[i] - 'A' + 'a' : s[i];
            //s[i] = tolower(s[i]);
            /*
            if(s[i] >= 'A' && s[i] <= 'Z'){
                s[i] = s[i] - 'A' + 'a';
            }
            */
        }
        return s;
    }
};
